core/src/main/java/bio/terra/pearl/core/service/address/AddressValidationClientStub.java

```diff
@@ -1,36 +1,44 @@
 package bio.terra.pearl.core.service.address;
 
+import bio.terra.pearl.core.model.address.AddressComponent;
 import bio.terra.pearl.core.model.address.AddressValidationResultDto;
-import bio.terra.pearl.core.model.participant.MailingAddress;
+import bio.terra.pearl.core.model.address.MailingAddress;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.http.HttpStatusCode;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.UUID;
 
 @Component
 @Slf4j
 public class AddressValidationClientStub implements AddressValidationClient {
 
     private static final String BAD_ADDRESS_INDICATOR = "BAD";
     private static final String IMPROVABLE_ADDRESS_INDICATOR = "IMPROVABLE";
+    private static final String INFERRED_ADDRESS_INDICATOR = "INFERRED";
     private static final String ERROR_INDICATOR = "ERROR";
+    private static final String VACANT_INDICATOR = "VACANT";
 
     /**
-     * Stubbed client allows you to test various possible states of address validation.
+     * Stubbed client allows you to test various possible states of address validation. In all cases, if you put
+     * "VACANT" in street1, the address will be considered vacant.
      * <br>
      * State 1: Invalid address
-     * - Put the word "BAD" in street1, e.g. 123 Bad St
+     * - Put the word "BAD" in street1, e.g. 123 Bad St, or leave it empty
      * - All the tokens in street1 will be considered unresolved
-     * - Example missing component data will be returned
+     * - Example missing component data will be returned based on simple rules
      * - Real-world responses are not guaranteed have missingComponents or unresolvedTokens
      * <br>
      * State 2: Improvable address
      * - Put the word "IMPROVABLE" in street1, e.g. 415 IMPROVABLE St
      * - Returns valid = true with a suggested address of 415 Main St
      * <br>
+     * State 3: Required inference
+     * - Put the word "INFERENCE" in street1, e.g. 415 INFERENCE St
```

Comment:
I think this should read:

```suggestion
     * - Put the word "INFERRED" in street1, e.g. 415 INFERRED St
```